### PR TITLE
Fixed Shuffle/Jump and enhanced Back

### PR DESCRIPTION
**Pull request by CanadaBry**
## Changes
**Fixed:** jump and shuffle functionality
- Will properly shuffle the first song, and jump to the correct song

**Enhanced:** back functionality to act like traditional playlist.
- No longer has a circular back, when using back current song is added to top of the queue

**Enhanced:** looping track logic to allow tracks to be skipped/jumped.
**Fixed**: currentTrack being added to previousTrack at both the start and end of a song. Now only added at the beginning.

### CODE REVIEW

**Comment by CanadaBry** [comment]
Been playing with discord-music-bot and this is just a batch of fixes/enhancements I've made for the version I self-hosting.

File: src/Structures/Queue.ts [hunk 1]
```diff
@@ -175,14 +176,14 @@ class Queue<T = unknown> {
...
             this._filtersUpdate = false;
+            this._trackSkipped = false;
         });
 
         this.connection.on("finish", async (resource) => {
             if (this.#watchDestroyed(false)) return;
             this.playing = false;
             if (this._filtersUpdate) return;
             this._streamTime = 0;
-            if (resource && resource.metadata) this.previousTracks.push(resource.metadata);
```
Comment (DevAndromeda) [inline src/Structures/Queue.ts:185]: why did you remove this?

**Comment by DevAndromeda** [comment]
@CanadaBry this needs a rebase

**Comment by CanadaBry** [comment]
I removed that because the track was pushed onto previousTracks when it starts the song. I am busy this weekend, but can rebase on Monday and give more details

File: src/Structures/Queue.ts (hunk 1 above)
Comment (CanadaBry) [reply src/Structures/Queue.ts:187]: On line 603 (in my branch) in the play function:
```javascript
        if (!options.filtersUpdate) {
            this.previousTracks = this.previousTracks.filter((x) => x.id !== track.id);
            this.previousTracks.push(track);
        }
```
There was a duplication of the track being pushed onto previousTracks. I removed the line to rectify this issue. It's possible to have removed the one on line 603 and kept the one I removed in my branch, but some existing functionality would need to be updated to account for the current song not existing in either previousTracks or queue until it finished.
